### PR TITLE
Add forgot password functions to SKYAuthContainer

### PR DESCRIPTION
**Pull request by Steven-Chan**
connect #110
SkygearIO/features#70

### CODE REVIEW

File: SKYKit.podspec [hunk 1]
```diff
@@ -12,8 +12,6 @@ Pod::Spec.new do |s|
 
   s.platform     = :ios, '7.0'
 
-  s.default_subspec = 'Core'
```
Comment (rickmak) [inline SKYKit.podspec]: What will happen without default subspec?

**Comment by rickmak** [comment]
Are users required to specific all subspec in their pod file?

```
pod "SKYKit", :path => "../"
pod "SKYKit/Facebook", :path => "../"
pod "SKYKit/ForgotPassword", :path => "../"
```

Seem not a good idea?

File: SKYKit.podspec (hunk 1 above)
Comment (Steven-Chan) [inline SKYKit.podspec]: Oh, my bad, I didn't see this line actually.

Base on the discussion on https://github.com/SkygearIO/skygear-SDK-iOS/pull/97, I think I would include 'Core' and 'ForgotPassword' in default spec.

'SKYKit/Facebook' depends on 'FBSDKCoreKit', which I think is a bad idea to make skygear looks like it depends on facebook, given that most pod users would just `pod 'SKYKit'`.

Comment (Steven-Chan) [inline SKYKit.podspec]: > What will happen without default subspec?

For this question, if not specified, then all subspec would be included.
https://guides.cocoapods.org/syntax/podspec.html#default_subspecs

**Comment by Steven-Chan** [comment]
Relavent issue: #92

**Pull request merged**